### PR TITLE
drivers/sensor: add support to LSM6DSV16X IMU sensor

### PR DESCRIPTION
**Pull request by avisconti**
The LSM6DSV16X is a system-in-package featuring a 3-axis digital accelerometer and a 3-axis digital gyroscope for industrial and IoT solutions. The LSM6DSV16X embeds advanced dedicated features such as a finite state machine (FSM) for configurable motion tracking and a machine learning core (MLC) for context awareness.

https://www.st.com/en/mems-and-sensors/lsm6dsv16x.html

This driver is based on stmemsc HAL i/f v2.02
It follows the inclusive language strategy adopted in #56800

### CODE REVIEW

File: drivers/sensor/lsm6dsv16x/lsm6dsv16x.h [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+ * https://www.st.com/resource/en/datasheet/lsm6dsv16x.pdf
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_LSM6DSV16X_LSM6DSV16X_H_
+#define ZEPHYR_DRIVERS_SENSOR_LSM6DSV16X_LSM6DSV16X_H_
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/types.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/spi.h>
```
Comment (MaureenHelm) [inline drivers/sensor/lsm6dsv16x/lsm6dsv16x.h:17]: You're including this header twice, remove this one

File: drivers/sensor/lsm6dsv16x/lsm6dsv16x_trigger.c [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+	if (enable) {
+		int16_t buf;
+
+		/* dummy read: re-trigger interrupt */
+		lsm6dsv16x_temperature_raw_get(ctx, &buf);
+	}
+
+	/* set interrupt (TEMP DRDY interrupt is only on INT2) */
+	if (cfg->drdy_pin == 1)
+		return -EIO;
```
Comment (MaureenHelm) [inline drivers/sensor/lsm6dsv16x/lsm6dsv16x_trigger.c:42]: wrap in braces

File: tests/drivers/build_all/sensor/i2c.dtsi [hunk 3]
```diff
@@ -728,3 +728,9 @@ test_i2c_lsm6dso16is: lsm6dso16is@6f {
 	reg = <0x6f>;
 	irq-gpios = <&test_gpio 0 0>;
 };
+
+test_i2c_lsm6dsv16x: lsm6dsv16x@6b {
```
Comment (MaureenHelm) [inline tests/drivers/build_all/sensor/i2c.dtsi:732]: Update the address

File: tests/drivers/build_all/sensor/spi.dtsi [hunk 4]
```diff
@@ -388,3 +388,10 @@ test_spi_lsm6dso16is: lsm6dso16is@2f {
 	spi-max-frequency = <0>;
 	irq-gpios = <&test_gpio 0 0>;
 };
+
+test_spi_lsm6dsv16x: lsm6dsv16x@2f {
```
Comment (MaureenHelm) [inline tests/drivers/build_all/sensor/spi.dtsi:392]: update the address

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Some small nits, but otherwise LGTM

File: drivers/sensor/lsm6dsv16x/lsm6dsv16x.h (hunk 1 above)
Comment (avisconti) [reply drivers/sensor/lsm6dsv16x/lsm6dsv16x.h]: thx

File: drivers/sensor/lsm6dsv16x/lsm6dsv16x_trigger.c (hunk 2 above)
Comment (avisconti) [reply drivers/sensor/lsm6dsv16x/lsm6dsv16x_trigger.c:42]: ack

File: tests/drivers/build_all/sensor/i2c.dtsi (hunk 3 above)
Comment (avisconti) [reply tests/drivers/build_all/sensor/i2c.dtsi]: ack

File: tests/drivers/build_all/sensor/spi.dtsi (hunk 4 above)
Comment (avisconti) [reply tests/drivers/build_all/sensor/spi.dtsi]: ack, thx

**Review by MaureenHelm — APPROVED** [review]

**Review by teburd — APPROVED** [review]
Looks great

**Pull request merged**